src/CppParser/Bootstrap/Utils.cs

```diff
@@ -0,0 +1,136 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using CppSharp;
+using CppSharp.AST;
+using CppSharp.Types;
+using CppAbi = CppSharp.Parser.AST.CppAbi;
+
+namespace CppSharp.Parser.Bootstrap
+{
+    public static class ClassExtensions
+    {
+        //FIXME: Multiple inheritance
+        public static bool isBaseOf (this Class baseclass, Class subclass)
+        {
+            if (subclass == null)
+                return false;
+            return subclass == baseclass || baseclass.isBaseOf (subclass.BaseClass);
+        }
+    }
+
+    abstract class ASTGenerator : TextGenerator
```

Comment:
Please put access specifiers before classes. Also, classes are internal by default and they'd better be public unless there's a very good reason not to.